backend/app/api/endpoints/recommendation.py

```diff
@@ -0,0 +1,32 @@
+from fastapi import APIRouter, HTTPException
+
+import app.use_case.recommendation as recommendation
+from app.schema.recommendation import FreeFormatQuery, StructuredQuery
+
+router = APIRouter(
+    prefix="/recommendation",
+    tags=["generate_recommendation"],
+)
+
+
+@router.post("/structured-format")
```

Comment:
Nit: I think the standard method for query should be get, and then we can use all the query params. In this case if you find that raises the overhead for later works then feel free to pass.

Reply:
The get method limits number of character can be sent to backend so I afraid there is case we run out of character (although it is unlikely).